lab-yunjoo/lib/task-crud.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+const debug = require('debug')('rito:note-crud');
+const mongoose = require('mongoose');
+mongoose.Promise = require('bluebird');
+const Task = require('../model/task');
+const noteCrud = require('./note-crud');
+const AppError = require('./app-error');
+
+exports.createTask = function(reqBody){
+  debug('creating task');
+  return new Promise((resolve, reject) => {
+    if(! reqBody.noteId)
+      return reject(AppError.error400('tasks require noteID'));
+    if(! reqBody.desc)
+      return reject(AppError.error400('tasks require a desc'));
+    const task = new Task(reqBody);
+    task.save()
+    .then(task => resolve(task))
+    .catch(err => reject(err));
+  });
+};
+
+exports.fetchTaskByNoteId = function(noteId){
+  return new Promise((resolve, reject) => {
+    noteCrud.fetchNote({_id: noteId})
+    .then(note =>Task.find({noteId: note._id}))
+    .then(tasks => resolve(tasks))
+    .catch(err => reject(err));
+  });
+};
+exports.fetchTaskByTaskId = function(id){
```

Comment:
This seems weird to me. The function is called fetchTaskByTaskId, but the argument that you put in is used as a note id. How does that work?